app/views/posts/show.html.erb

```diff
@@ -1,6 +1,11 @@
 <section class="container">
     <div class="flex-col post-card">
-        <h1><%= @post.title %></h1>
+        <div class="post-title">
+            <h1><%= @post.title %></h1>
+            <% if @post && can?(:destroy, @post) %>
+                <%= button_to "Delete", user_post_path(user_id: @user.id, id: @post.id), method: :delete, data: { confirm: "Are you sure?" } %>
+            <% end %>
+        </div>
         <span >by <%= @user.name %></span>
         <span >Comments : <%= @post.comments_counter || 0 %>, Likes : <%= @post.likes_counter || 0 %></span>
 
```

Comment:
![image](https://github.com/Mofuhidy/blog-app/assets/93039359/5eaeb93e-57a1-41b3-8264-2ca3b3bf4097)

**[OPTIONAL BUT HIGHLY RECOMMENDED]**

- [ ] It would be nice if you fix the error coming from the likes controller because when I try to like someone else's post it creates the like in the database and throws this error 😨 Kindly fix this error to make your app error-free 👍

Reply:
Okay @HAMMAS-SALEEM  Thanks a lot